### PR TITLE
Fix `aa init`

### PR DESCRIPTION
**Pull request by vmax**

### CODE REVIEW

File: tools/aa/commands/InitCommand.java [hunk 1]
```diff
@@ -33,23 +36,39 @@
...
 
     try {
+      String remoteRepoPath = remoteRepoUrl.get();
       Config config =
           Config.newBuilder()
               .setBasePath(basePath.get())
-              .setRemoteRepoUrl(remoteRepoUrl.get())
+              .addRemoteRepo(
+                  RemoteRepo.newBuilder()
+                      .setId(getNameFromRemoteUrl(remoteRepoPath))
```
Comment (oferb) [inline tools/aa/commands/InitCommand.java]: Good idea to get that automatically

File: tools/aa/commands/InitCommand.java [hunk 2]
```diff
@@ -33,23 +36,39 @@
...
+    String path = new URI(remoteRepoUrl.get()).getPath();
+    return path.substring(path.lastIndexOf("/") + 1).replace(".git", "");
+  }
+
   @Override
   public void run(String[] args) {
     Flags.parse(args, InitCommand.class.getPackage());
 
     try {
+      String remoteRepoPath = remoteRepoUrl.get();
```
Comment (oferb) [inline tools/aa/commands/InitCommand.java]: I would switch - 
`String remoteRepoId = getNameFromRemoteUrl(remoteRepoUrl.get());`
Then,
```
RemoteRepo.newBuilder()
    .setId(remoteRepoId)
    .setUrl(remoteRepoUrl.get()))
```

File: tools/aa/AaTool.java [hunk 3]
```diff
@@ -17,20 +17,17 @@
...
-import java.text.ParseException;
-import java.util.Arrays;
+import com.google.startupos.tools.aa.commands.CommandsComponent;
+import com.google.startupos.tools.aa.commands.DaggerCommandsComponent;
 import java.util.HashMap;
 
 /** aa tool. */
 public class AaTool {
   private static HashMap<String, AaCommand> commands = new HashMap<>();
+  private static CommandsComponent commandsComponent = DaggerCommandsComponent.create();
```
Comment (oferb) [inline tools/aa/AaTool.java]: Would be cleaner to define one component for the AaTool, here as a nested interface, similar to how it's done at `LocalServer`:
https://github.com/google/startup-os/blob/master/tools/local_server/LocalServer.java#L95-L99

Then, `AaTool` would get `InitCommand` in its constructor.

File: tools/aa/BUILD [hunk 4]
```diff
@@ -19,11 +19,13 @@ java_binary(
     main_class = "com.google.startupos.tools.aa.AaTool",
     deps = [
         ":config_java_proto",
+        "//common",
+        "//common:dagger_with_annotation_processor",
+        "//common/flags",
```
Comment (oferb) [inline tools/aa/BUILD]: We probably don't need `"//common/flags",` in `AaTool` anymore..

Comment (vmax) [reply tools/aa/BUILD]: @oferb `InitCommand` does use it and it's linked to `aa_tool` as well

Comment (oferb) [reply tools/aa/BUILD]: Ohh right..

**Pull request merged**